fleetspeak/src/client/https/polling_test.go

```diff
@@ -427,46 +425,59 @@ func (hp *httpsProxy) close() {
 // Handles the request by hijacking the HTTP connection, opening a second connection to the host of
 // the request and copying data between the two connections.
 func (hp *httpsProxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	t := hp.t
+	returnError := func(err error) {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+
 	d, err := httputil.DumpRequest(r, false)
 	if err != nil {
-		t.Fatal(err)
+		returnError(err)
+		return
 	}
-	t.Logf("Got proxy request: %s.", string(d))
+	log.Infof("Got proxy request: %s.", string(d))
+
 	if r.Method != http.MethodConnect {
-		t.Fatalf("Proxy received invalid method: %s.", r.Method)
+		http.Error(w, fmt.Sprintf("Proxy received invalid method: %s.", r.Method), http.StatusInternalServerError)
+		return
 	}
 	conn, err := net.Dial("tcp", r.Host)
 	if err != nil {
-		t.Fatal(err)
+		returnError(err)
+		return
 	}
+	defer conn.Close()
 	w.WriteHeader(http.StatusOK)
 	hijacker, ok := w.(http.Hijacker)
 	if !ok {
-		t.Fatal("Failed to hijack HTTP connection.")
+		log.Error("Failed to hijack HTTP connection.")
+		return
 	}
 	httpConn, _, err := hijacker.Hijack()
 	if err != nil {
-		t.Fatal(err)
+		log.Error(err)
```

Comment:
I think adding a bit of context would help here. The error returned might not have enough information to allow one to deduce that the error happened while trying to hijack a connection.

Reply:
Done.